blockchain/chain/src/store/chain_store.rs

```diff
@@ -731,14 +731,13 @@ where
 }
 
 /// Returns the genesis block from storage.
-pub fn genesis<DB>(db: &DB) -> Result<Option<BlockHeader>, Error>
+pub fn genesis<DB>(db: &DB) -> Result<BlockHeader, Error>
```

Comment:
This function should only be used from the `ChainStore`. Let's move it into `ChainStore::genesis`.

Reply:
And let's do the same with `set_genesis`. That function should also only be used through the `ChainStore`.

Reply:
Moved